src/annotation.js

```diff
@@ -0,0 +1,117 @@
+module.exports = function(Chart) {
+	var chartHelpers = Chart.helpers;
+
+	var helpers = require('./helpers.js')(Chart);
+	var events = require('./events.js')(Chart);
+
+	var annotationTypes = Chart.Annotation.types;
+
+	return {
+		beforeInit: function(chartInstance) {
+			var chartOptions = chartInstance.options;
+
+			// Initialize chart instance plugin namespace
+			var ns = chartInstance.annotation = {
+				elements: {},
+				config: helpers.initConfig(chartOptions.annotation || {}),
+				onDestroy: [],
+				firstRun: true
+			};
+
+			ns[ns.config.drawTime] = function(easingDecimal) {
+				helpers.elements(chartInstance).forEach(function(element) {
+					element.transition(easingDecimal).draw();
+				});
+			};
+
+			// Add the annotation scale adjuster to each scale's afterDataLimits hook
+			chartInstance.ensureScalesHaveIDs();
+			chartHelpers.each(chartOptions.scales.xAxes, function(axisOptions) {
+				helpers.decorate(axisOptions, 'afterDataLimits', function(previous, scale) {
+					if (previous) previous(scale);
+					Chart.Annotation.ScaleAdjuster(scale);
+				});
+			});
+			chartHelpers.each(chartOptions.scales.yAxes, function(axisOptions) {
+				helpers.decorate(axisOptions, 'afterDataLimits', function(previous, scale) {
+					if (previous) previous(scale);
+					Chart.Annotation.ScaleAdjuster(scale);
+				});
+			});
+		},
+		beforeUpdate: function(chartInstance) {
+			var ns = chartInstance.annotation;
+
+			if (!ns.firstRun) {
+				ns.config = helpers.initConfig(chartInstance.options.annotation || {});
+			} else {
+				ns.firstRun = false;
+			}
+
+			var elementIds = [];
+
+			// Add new elements, or update existing ones
+			ns.config.annotations.forEach(function(annotation) {
+				var element;
+				if (typeof annotation._id === 'undefined' && !!annotationTypes[annotation.type]) {
+					annotation._id = helpers.objectId();
+					var cls = annotationTypes[annotation.type];
```

Comment:
`jshint` doesn't complain. I'd rather have it lowercase, because convention is that newable _classes_ are capital. `cls` is a variable class reference, not the class definition itself. I think this subtle difference makes the code more readily understandable.